controller/src/main/java/io/pravega/controller/server/SegmentHelper.java

```diff
@@ -605,7 +622,9 @@ private void closeConnection(Reply reply, RawClient client) {
     }
 
     private <T extends Request & WireCommand> CompletableFuture<Reply> sendRequest(RawClient connection, long requestId, T request) {
-        return connection.sendRequest(requestId, request)
+        CompletableFuture<Reply> future = connection.sendRequest(requestId, request);
+        Futures.addTimeout(future, timeout.get(), executorService);
```

Comment:
`Futures.futureWithTimeout(() -> connection.sendRequest(...), ...)` ?

That will preclude the need for `Futures.addTimeout`